feat: doc update with queue svc (#367)
screwdriver-cd/screwdriver-cd.github.io: docs/cluster-management/configure-queue-service.md

---
layout: main
title: Configuring the Queue Service
category: Cluster Management
menu: menu
toc: 
    - title: Managing the Queue Service
      url: "#managing-the-queue-service"
      active: true
    - title: Packages
      url: "#packages"
    - title: Configuration
      url: "#configuration"
---
# Managing the Queue Service

## Packages

Like the other services, the API is shipped as a [Docker image](https://hub.docker.com/r/screwdrivercd/queue-service/) with port 80 exposed.

```bash
$ docker run -d -p 7000:80 screwdrivercd/queue-service:latest

> $ open http://localhost:7000

Reviewer: I think this should be `9003` which is consistent to 

```
 httpd:
  port: 9003
  host: 0.0.0.0
  uri: http://YOUR_IP:9003
```

from the `queue-service` config

Author: > I think this should be `9003` which is consistent to
> 
> ```
>  httpd:
>   port: 9003
>   host: 0.0.0.0
>   uri: http://YOUR_IP:9003
> ```
> 
> from the `queue-service` config

Updated to 80 there, in PR.

| URI       | http://localhost:80 | Externally routable URI (usually your load balancer or CNAME)                                                                                                                                              |
| HTTPD_TLS | false               | SSL support; for SSL, replace `false` with a JSON object that provides the options required by [`tls.createServer`](https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener) |

```yaml
# config/local.yaml
httpd:
    port: 443
    host: 0.0.0.0
    uri: https://localhost
    tls:
        key: |
            PRIVATE KEY HERE
        cert: |
            YOUR CERT HERE
```

### Configure Redis Queue

Configure some settings about storing Build Artifacts.

| Key                | Required            |  Default              | Description       |
|:-------------------|:---------------------|:----------------------------------------------------|
| REDIS_HOST          | Yes                  | 127.0.0.1            | Redis host                  |

> | REDIS_PORT          | Yes                  | 9999                 | Redis port                  |

Reviewer: Could we change to redis default port `6379`? Its easier to follow and search.

| REDIS_PASSWORD      | Yes                  | a-secure-password    | Redis password              |
| REDIS_TLS_ENABLED   | No                   | false                | Redis tls enabled           |
| REDIS_DB_NUMBER     | No                   | 0                    | Redis db number             |
| REDIS_QUEUE_PREFIX  | No                   | ''                   | Redis queue prefix          |
```yaml
# config/local.yaml
queue:
  redisConnection:
    host: "127.0.0.1"

>     port: 9999

Reviewer: Could we change to redis default port `6379`? Its easier to follow and search.

---
layout: main
title: Configuring the Queue Service
category: Cluster Management
menu: menu
toc: 
    - title: Managing the Queue Service

>       url: "#managing-the-queue-svc"

Reviewer 2: ```suggestion
      url: "#managing-the-queue-service"
```

      active: true
    - title: Packages
      url: "#packages"
    - title: Configuration
      url: "#configuration"
---
# Managing the Queue Service

## Packages

Like the other services, the API is shipped as a [Docker image](https://hub.docker.com/r/screwdrivercd/queue-service/) with port 80 exposed.

```bash
$ docker run -d -p 7000:80 screwdrivercd/queue-service:latest
$ open http://localhost:7000
```

> Our images are tagged for their version (eg. `1.2.3`) as well as a floating `latest` and `stable`. Most installations should be using `stable` or the fixed version tags.

Reviewer 2: ```suggestion
Our images are tagged with the version (eg. `v1.2.3`) as well as a floating tag `latest` and `stable`. Most installations should be using `stable` or the fixed version tags.
```

## Configuration

Screwdriver already [defaults most configuration](https://github.com/screwdriver-cd/queue-service/blob/master/config/default.yaml), but you can override defaults using a `config/local.yaml` or environment variables. All the possible environment variables are [defined here](https://github.com/screwdriver-cd/queue-service/blob/master/config/custom-environment-variables.yaml).

### Authentication

Configure the validation of incoming JWTs from the API.

> | Key                   | Default | Description                                                                                           |

Reviewer 2: Do you mean `Required` rather than `Default`?

| URI       | http://localhost:80 | Externally routable URI (usually your load balancer or CNAME)                                                                                                                                              |
| HTTPD_TLS | false               | SSL support; for SSL, replace `false` with a JSON object that provides the options required by [`tls.createServer`](https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener) |

```yaml
# config/local.yaml
httpd:
    port: 443
    host: 0.0.0.0
    uri: https://localhost
    tls:
        key: |
            PRIVATE KEY HERE
        cert: |
            YOUR CERT HERE
```

### Configure Redis Queue

Configure some settings about storing Build Artifacts.

| Key                | Default              | Description                                         |
|:-------------------|:---------------------|:----------------------------------------------------|
| REDIS_HOST    | 127.0.0.1      | Redis host                        |

> | REDIS_PORT    | 9999           | Redis port                        |

Reviewer 2: are you missing some vars?
https://github.com/screwdriver-cd/queue-service/blob/master/config/custom-environment-variables.yaml#L216-L220

Author: Updated vars
